packages/snowpack/src/build/build-import-proxy.ts

```diff
@@ -65,7 +67,20 @@ export function wrapHtmlResponse({
   code = code.replace(/\/?%PUBLIC_URL%\/?/g, config.buildOptions.baseUrl);
 
   if (hmr) {
-    code += `<script type="module" src="${getMetaUrlPath('hmr.js', isDev, config)}"></script>`;
+    const hmrScript = `<script type="module" src="${getMetaUrlPath(
+      'hmr.js',
+      isDev,
+      config,
+    )}"></script>`;
+
+    const closingBodyMatch = code.match(CLOSING_BODY_TAG);
+    if (closingBodyMatch && closingBodyMatch.length === 1) {
```

Comment:
I don't think match returns a full array of matches, that would be `matchAll` but that's not supported in Node v10.

Code golf time!

```
const firstMatch = code. indexOf('</body>');
const lastMatch = code.lastIndexOf('</body>');
if (firstMatch > -1 && firstMatch === lastMatch) {
  // inject it at that index
}
```

Reply:
Shoot but I don’t think that handles case-insensitivity (`</BODY>`) or whitespace (`</ body >`). I just tested it / looked it up and `code.match()` does return all matched results on Node 10 and Node 12. You’re right `matchAll()` is only v12+, but that’s only needed when we’re doing sub-matches with capture groups, which we don’t need here.

Reply:
TIL! that's great then, ignore me :)

Reply:
No no thanks for bringing it up! I can never remember `match()` vs `matchAll()` so I’m glad you raised it and got me to double-check.

Reply:
Maybe something like this `/<\/body>(?![\s\S]*<\/body>[\s\S]*$)/ig` would get the last match?